lib/xdparser2.ts

```diff
@@ -676,92 +676,104 @@ function parseSplitsFromAnswer(answerWithSplits: string, splitCharacter?: string
 
 export function inlineMarkdownParser(str: string): MDClueComponent[] {
   const components: MDClueComponent[] = []
-  let token = ""
-  let mode: MDClueComponent[0] = "text"
-  let linkText = ""
-
-  const pushText = () => {
-    if (token.length > 0) {
-      components.push(["text", token])
-      token = ""
+  const mdTokens = ["**", "\\\\", "~~", "[", "]", "(", ")"]
+  const getMDType = (operator: string) => {
+    switch (operator) {
+      case "**": return "bold"
+      case "\\\\": return "italics"
+      case "~~": return "strike"
     }
+    throw Error("Operator: " + operator + " not handled")
   }
 
-  for (let index = 0; index < str.length; index++) {
-    const prevLetter = index > 0 ? str.slice(index - 1, index) : ""
-    const letter = str.slice(index, index + 1)
-    const nextLetter = str.slice(index + 1, index + 2)
-
-    const otherMDKeys = ["*", "/", "~", "["]
-    if (letter === "\\" && otherMDKeys.includes(nextLetter)) {
-      index++
-      token += nextLetter
-      continue
-    } else if (mode === "text" && prevLetter !== "\\") {
-      if (letter === "[") {
-        mode = "link"
-        pushText()
-        continue
+  const escapeCharacter = "^"
+  let textSlice = ''
+  // stack of [index, md operators like "**" etc]
+  const stack: Array<[number, string]> = []
+  for (let index = 1; index < str.length; index++) {
```

Comment:
If you are going through the effort or rewriting this parser, go one more step further and use some functions. It will help the next person understand the purpose of different parts of this code.